### PR TITLE
[TE] Add buffer  to holiday pipeline

### PR DESCRIPTION
**Pull request by npawar**
The holiday pipeline adds a buffer of 2 days prior, to the time range it is investigating. This is done because an issue happening today could be because of a holiday that just happened a day or 2 ago.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java [hunk 1]
```diff
@@ -55,15 +60,36 @@ public HolidayEventsPipeline(String outputName, Set<String> inputNames, Map<Stri
...
-
-    EventFilter filter = new EventFilter();
-    filter.setEventType(EventType.HOLIDAY.toString());
-    filter.setStartTime(current.getStart());
-    filter.setEndTime(current.getEnd());
+    long currentStart = new DateTime(current.getStart()).minusDays(2).getMillis();
+    long currentEnd = current.getEnd();
+    long baselineStart = new DateTime(currentStart).minusDays(BASELINE_OFFSET).getMillis();
+    long baselineEnd = new DateTime(currentEnd).minusDays(7).getMillis();
+    TimeRangeEntity baseline = TimeRangeEntity.fromRange(1.0, TimeRangeEntity.TYPE_BASELINE, baselineStart, baselineEnd);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: Is this left over from debugging?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java [hunk 2]
```diff
@@ -55,15 +62,36 @@ public HolidayEventsPipeline(String outputName, Set<String> inputNames, Map<Stri
...
-
-    EventFilter filter = new EventFilter();
-    filter.setEventType(EventType.HOLIDAY.toString());
-    filter.setStartTime(current.getStart());
-    filter.setEndTime(current.getEnd());
+    long currentStart = new DateTime(current.getStart()).minusDays(HOLIDAY_DAYS_BUFFER).getMillis();
+    long currentEnd = current.getEnd();
+    long baselineStart = new DateTime(currentStart).minusDays(BASELINE_OFFSET).getMillis();
+    long baselineEnd = new DateTime(currentEnd).minusDays(BASELINE_OFFSET).getMillis();
+    TimeRangeEntity baseline = TimeRangeEntity.fromRange(1.0, TimeRangeEntity.TYPE_BASELINE, baselineStart, baselineEnd);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: I don't quite understand the reasoning behind building a new baseline entity. There's a baseline TimeRangeEntity in the search context.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java (hunk 1 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: Change is intentional, shabby numbers are not. Fixed this to use constants. Added to javadoc to explain why this is done

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java (hunk 2 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: Is that baseline mandatory?

Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: Fixed.

Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsPipeline.java]: Yes

**Comment by apucher** [comment]
LGTM

**Pull request merged**